x-pack/plugins/maps/public/actions/layer_actions.ts

```diff
@@ -539,3 +539,22 @@ export function setHiddenLayers(hiddenLayerIds: string[]) {
     }
   };
 }
+
+export function setAreTilesLoaded(layerId: string, areTilesLoaded: boolean) {
+  return (
```

Comment:
No need to make action a thunk. LayerId is passed in so you don't need to getLayerById just to call layer.getId() again. Method can just be 

```
export function setAreTilesLoaded(layerId: string, areTilesLoaded: boolean) {
  return {
    type: UPDATE_LAYER_PROP,
    id: layerId,
    propName: '__areTilesLoaded',
    newValue: areTilesLoaded,
  };
}
```